src/wp-includes/rest-api/endpoints/class-wp-rest-comments-controller.php

```diff
@@ -151,6 +145,36 @@ public function get_items_permissions_check( $request ) {
 						array( 'status' => rest_authorization_required_code() )
 					);
 				}
+
+				if ( $post && $is_note && ! $this->check_post_type_supports_notes( $post->post_type ) ) {
+					if ( current_user_can( get_post_type_object( $post->post_type )->cap->edit_posts ) ) {
```

Comment:
Is this right? Why not use the same check as is used below?

```suggestion
					if ( current_user_can( 'edit_post', $post->ID ) {
```

Reply:
Thanks. Accepting and reviewing entire PR again.

Reply:
@adamsilverstein @westonruter The check below is when no post is specified so the `post` post type is used as the detault. When a post ID is specified the post type can be a CPT which may have custom capabilities so we need to check those for accuracy.

Reply:
But aren't those custom capabilities automatically mapped when using the `edit_post` capability?

Reply:
@westonruter Sorry, I misread the change. You're right, it's all good.